### PR TITLE
Pretty substantial re-write / re-structure

### PR DESCRIPTION
**Pull request by B3none**
Worth noting that this will require a major version bump.

### CODE REVIEW

**Comment by cjmaxik** [comment]
What's the point of update? Please explain this changes.

**Comment by B3none** [comment]
Pushing the version to PHP 7.1 syntax so that people using the client are on the latest supported version of PHP.

https://www.php.net/supported-versions.php

I've also altered a little bit of the logic to prevent duplicate code by implementing an abstract class for GroupedModels.

File: Tests/ClientTest.php [hunk 1]
```diff
@@ -3,75 +3,81 @@
...
+use TruckersMP\Models\PlayerModel;
 
-class ClientTest extends \PHPUnit_Framework_TestCase
+class ClientTest extends TestCase
 {
-    private $testAccount = 585204;
+    /**
+     * @var int
+     */
+    protected $testAccount = 162950;
```
Comment (cjmaxik) [inline Tests/ClientTest.php]: I see what you did there but this is not reliable at all. We might issue a ban on our test account. Also, there is no check if bans are hidden on account.

File: composer.json [hunk 2]
```diff
@@ -21,13 +21,21 @@
...
+    "php": ">=7.1.0",
     "nesbot/carbon": "^1.21",
     "php-http/message": "^1.2",
-    "guzzlehttp/psr7": "^1.3"
+    "guzzlehttp/psr7": "^1.3",
+    "leedavis81/vent": "dev-master",
+    "ext-json": "*"
   },
   "require-dev": {
     "phpunit/phpunit": "^5.3",
```
Comment (cjmaxik) [inline composer.json]: PHPUnit 5.3 is no longer supported (_Support ended on February 2, 2018_). Use at least v7.

File: Tests/ClientTest.php (hunk 1 above)
Comment (B3none) [reply Tests/ClientTest.php]: I'll revert it to the test account and so sanity checks on the data. I agree that this is not the best idea. I'll implement the handling for no bans too and test accordingly.

File: composer.json (hunk 2 above)
Comment (B3none) [reply composer.json]: Will upgrade, totally missed that one.

File: composer.json [hunk 3]
```diff
@@ -21,16 +21,24 @@
...
     }
   ],
   "require": {
-    "php": ">=5.6.0",
+    "php": ">=7.1.0",
     "nesbot/carbon": "^1.21",
     "php-http/message": "^1.2",
-    "guzzlehttp/psr7": "^1.3"
+    "guzzlehttp/psr7": "^1.3",
+    "leedavis81/vent": "dev-master",
```
Comment (cjmaxik) [inline composer.json]: Please use other solution since this is old bundle and there is no stable version.

**Review by cjmaxik — CHANGES_REQUESTED** [review]
Please check all the dependencies since they are not using latest and/or stable version.

**Comment by B3none** [comment]
Updates:
- Use phpunit `v7.5.8`
- Implement the Rules endpoint
- Update unit test

**Comment by B3none** [comment]
@cjmaxik This is all working as to be expected now. Please check the build to see why it failed (not an error with the code).

File: .travis.yml [hunk 4]
```diff
@@ -7,8 +7,6 @@ cache:
     - $HOME/.composer/cache
 
 php:
-  - 5.6
-  - 7.0
   - 7.1
```
Comment (cjmaxik) [inline .travis.yml]: Add 7.2 and (maybe) 7.3, remove hhvm

Comment (cjmaxik) [inline .travis.yml:10]: Add 7.2 and (maybe) 7.3, remove hhvm

File: Tests/ClientTest.php [hunk 5]
```diff
@@ -3,97 +3,128 @@
...
+use TruckersMP\Models\BanModel;
+use TruckersMP\Models\BansModel;
+use TruckersMP\Models\GameTimeModel;
+use TruckersMP\Models\PlayerModel;
+use TruckersMP\Models\RulesModel;
+
+class ClientTest extends TestCase
 {
-    private $testAccount = 585204;
+    const TEST_ACCOUNT = 585204;
```
Comment (cjmaxik) [inline Tests/ClientTest.php:16]: We should add a proper test account for production (maybe even several accounts for checking private states)

File: Tests/ClientTest.php [hunk 6]
```diff
@@ -3,97 +3,128 @@
...
      * @throws \Exception
      * @throws \Http\Client\Exception
      */
-    public function testServers()
+    public function testServers(): void
     {
         $servers = $this->client->servers();
 
-        $this->assertEquals($servers[0]->name, 'Europe 1');
+        $this->assertEquals($servers->getServers()[0]->getName(), 'Europe 1 [Simulation]');
```
Comment (cjmaxik) [inline Tests/ClientTest.php:77]: This should be streamlined for the cases if we will change server names.

File: src/Client.php [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+    /**
+     * Client constructor.
+     *
+     * @param array $config
+     * @param bool $secure
+     */
+    public function __construct(array $config = [], bool $secure = true)
+    {
+        $scheme = $secure ? 'https' : 'http';
```
Comment (cjmaxik) [inline src/Client.php:36]: Should the scheme be always secure?

**Comment by ShawnCZek** [comment]
Closing this pull request due to inactivity. Additionally, this was replaced and extended by the new pull request; #12.

**Pull request closed without merging**